### PR TITLE
Widget for Database.connect

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Use auto-scoping for Redshift, SQLServer and Snowflake.
image
- Fix for widgets on Header functions.
image
- Move `Snowflake_Details` to `Standard.Snowflake.Connection` namespace to make widgets work.
- Add widgets to `Snowflake_Details`.
image
- Typo fix for SQLServer SPI.
- Change SQLServer port to be an Integer and added default.
- Reordered parameters on SQLServer **(potentially breaking change)**.
- Added widgets to SQLServer.
image
- Added widget for JDBC options (as well as conversion from Vector to options).
image
- Added header alias to `use_first_row_as_names`.
- Added various aliases to `read` and `write`.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: std-bits/database/src/main/java/org/enso/database/DatabaseConnectionDetailsSPI.java [hunk 1]
```diff
@@ -27,6 +29,23 @@ public static String[][] get_default_constructors(boolean refresh) {
...
+   * Returns an array of all the types that implement the `DatabaseConnectionDetailsSPI` interface.
+   *
+   * @param refresh whether to refresh the list of types
+   * @return an array of all the types that implement the `DatabaseConnectionDetailsSPI` interface
+   */
+  public static Value[] get_types(boolean refresh) {
+    if (refresh) {
+      loader.reload();
+    }
+    return loader.stream().map(provider -> provider.get().getTypeObject()).toArray(Value[]::new);
```
Comment (GregoryTravis) [inline std-bits/database/src/main/java/org/enso/database/DatabaseConnectionDetailsSPI.java:42]: Does this iterate only through the SPI classes?

**Review by GregoryTravis — APPROVED** [review]

File: std-bits/database/src/main/java/org/enso/database/DatabaseConnectionDetailsSPI.java (hunk 1 above)
Comment (jdunkerley) [reply std-bits/database/src/main/java/org/enso/database/DatabaseConnectionDetailsSPI.java:42]: Yes just iterates over the SPIs for the DatabaseConnectionDetails.

The loader specifies which ones it finds in the type arg.

File: distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Connection_Options.enso [hunk 2]
```diff
@@ -12,3 +17,16 @@ type Connection_Options
     merge :  Vector -> Vector
     merge self base_options =
         base_options.filter x->(self.options.any (y->y.first==x.first) . not) + self.options
+
+    ## PRIVATE
+       Widget to design connection options.
+    default_widget : Widget
+    default_widget =
+        fqn = Meta.get_qualified_type_name Pair . drop (..Last 5)
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Connection_Options.enso:25]: I think I'd prefer to use
```suggestion
        fqn = Meta.get_qualified_type_name Pair . take (..Before_Last ".")
```
but that is quite subjective.

Comment (jdunkerley) [reply distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Connection_Options.enso:25]: Will tweak in my next PR.

**Review by radeusgd — APPROVED** [review]
Looks all good to me

**Pull request merged**